pom.xml

```diff
@@ -523,7 +523,7 @@
 				    -->
                                 </requireMavenVersion>
                                 <requireJavaVersion>
-                                    <version>[17.0.3,22.0)</version>
+                                    <version>[21.0.0,22.0)</version>
```

Comment:
This means we'll need to update the CLI and Support (and perhaps the UI, for openapi generation), to also use Java 21, doesn't it?

Reply:
Fair point, maybe I should actually be upgrading those projects first and then moving back to core.
Will leave the VM changes so our build runs on Java 21, same as our Docker container and split off tickets